tests/test_gdpr_api_delete.py

```diff
@@ -1,161 +1,225 @@
+from typing import Optional
+
 import pytest
+import requests_mock
+from django.conf import settings
 from django.contrib.auth import get_user_model
 from django.urls import reverse
+from rest_framework.test import APIClient
 
+from helsinki_gdpr.types import Error, ErrorResponse
 from tests.conftest import get_api_token_for_user_with_scopes
+from tests.factories import ExtraDataFactory
 
 from .models import Profile
 
 User = get_user_model()
 
 
-def test_delete_profile_dry_run_data(
-    true_value, api_client, profile, requests_mock, settings
+def do_delete(
+    user,
+    id_value,
+    scopes=(settings.GDPR_API_DELETE_SCOPE,),
+    query_params=None,
+    data=None,
+    url_id_param_name="pk",
 ):
-    auth_header = get_api_token_for_user_with_scopes(
-        profile.user, [settings.GDPR_API_DELETE_SCOPE], requests_mock
-    )
-    api_client.credentials(HTTP_AUTHORIZATION=auth_header)
-    response = api_client.delete(
-        reverse("helsinki_gdpr:gdpr_v1", kwargs={"pk": profile.id}),
-        data={"dry_run": true_value},
-        format="json",
-    )
+    api_client = APIClient()
+
+    with requests_mock.Mocker() as req_mock:
+        auth_header = get_api_token_for_user_with_scopes(user, scopes, req_mock)
+        api_client.credentials(HTTP_AUTHORIZATION=auth_header)
+
+        if query_params:
+            query = "?" + "&".join([f"{k}={v}" for k, v in query_params.items()])
```

Comment:
It could be nicer to use urllib when generating the url and its parameters. Even when this is "only" in the test code. What do you think?

Reply:
Yep, changed.